### PR TITLE
PHP8  support for SafeCurl

### PR DESCRIPTION
**Pull request by soorajfrancis**
Ref: - https://higherlogic.atlassian.net/browse/VNLA-2729

**Description**
Add Ability to support curl Handle

### CODE REVIEW

File: src/SafeCurl.php [hunk 1]
```diff
@@ -151,7 +151,8 @@ private function removeHeadersFromOutput(string $output, int $headersLength): st
      * @param resource|CurlHandler $curlHandle
      */
     public function setCurlHandle($curlHandle): void {
-        if (is_resource($curlHandle)){
+        //From php 8 curl sends out CurlHandle instance instead of resource
```
Comment (vanilla-dbarbier) [inline src/SafeCurl.php]:
```suggestion
        // From PHP 8, curl sends out CurlHandle instance instead of resource
```

**Review by vanilla-dbarbier — DISMISSED** [review]
Makes sense.
Made a wording suggestion.
However, CircleCI seems stuck.
Any way we could have the tests go through?

**Comment by soorajfrancis** [comment]
@vanilla-dbarbier  I don't seem to understand why the tests are not running. Might need help of Adam here to understand why it's not running at all.

File: src/SafeCurl.php [hunk 2]
```diff
@@ -249,6 +250,10 @@ public function setUrlValidator(UrlValidator $urlValidator): void {
      * @param resource $curlHandle
      */
     private function validateCurlHandle($curlHandle): void {
+
+        if(is_a($curlHandle, "CurlHandle")){
+            return ;
+        }
```
Comment (charrondev) [inline src/SafeCurl.php]:
```suggestion
        if(is_a($curlHandle, "CurlHandle")){
            return;
        }
```

**Review by charrondev — APPROVED** [review]

**Pull request merged**